starling/filters/FragmentFilter.hx

```diff
@@ -321,37 +322,33 @@ class FragmentFilter
         context.setVertexBufferAt(mVertexPosAtID, null);
         context.setVertexBufferAt(mTexCoordsAtID, null);
         context.setTextureAt(mBaseTextureID, null);
+        Starling.current.removeCurrentProgram(); 
 
         support.popMatrix();
         support.popMatrix3D();
-
+        
         if (intoCache)
         {
             // restore support settings
             support.projectionMatrix.copyFrom(projMatrix);
-            support.projectionMatrix3D.copyFrom(projMatrix3D);
+            support.projectionMatrix3D = projMatrix3D;
```

Comment:
If you change this, wouldn't it make sense to changet this for `projMatrix` in the line above as well? It would be also more consistent to similar calls in line 301 and 302.